src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -209,6 +211,7 @@ static int add_mount(
 
         log_debug("Adding %s: %s fstype=%s", where, what, fstype ?: "(any)");
 
+#if HAVE_LIBCRYPTSETUP
         if (streq_ptr(fstype, "crypto_LUKS")) {
```

Comment:
hmm, I think we should still do something here and generate a clear error message that we encountered an encrypted partition but we have no support for it.

And I am pretty sure we should hide that inside of add_cryptetup(), so that the ifdeffery is kept at the innermost encapsulation level. hence, please drop the `#if` here, and instead change add_cryptsetup() to fail with EOPNOTSUPP or so + a friendly log message if cryptsetup support is not enabled